src/main/kotlin/com/sourcegraph/cody/history/HistoryTree.kt

```diff
@@ -78,6 +78,17 @@ class HistoryTree(
         for (child in sorted) period.add(child)
         model.reload(period)
       }
+    } else {
+      val period =
+          root
+              .periods()
+              .flatMap { it.leafs() }
+              .find { it.chat.internalId == chat.internalId }!!
+              .parent as PeriodNode
+      val sorted = period.leafs().sortedByDescending { it.chat.getUpdatedTimeAt() }
```

Comment:
`period.leafs().sortedByDescending { it.chat.getUpdatedTimeAt() }` is now good candidate for a function (maybe smth like `Period.sortLeafs()`?).